app/controllers/posts_controller.rb

```diff
@@ -11,5 +12,23 @@ def show
     @user = User.find(user_id)
     @post = Post.find(post_id)
     @comments = Comment.where(post_id)
+    @recent_user = current_user
+  end
+
+  def new
+    @user = User.find(params[:user_id])
+    @post = Post.new
+    # redirect_to new_user_post_path(:user_id)
+  end
+
+  def create
+    user = current_user
+    @post = Post.new(
+      title: params[:title],
+      text: params[:text],
+      author_id: user.id
+    )
+    return unless @post.save
+    redirect_to user_posts_path
```

Comment:
- [ ] Ensuring your application creates a post will be great. It seems it is impossible for a user to create a post after you have created a page for the form when I try creating a post am getting this error in the terminal. 🔽 
<img width="746" alt="routing error" src="https://user-images.githubusercontent.com/32361963/230715157-1a1b3583-910a-4f2d-84a4-54f6630162f2.png">

Also, it is advisable to handle this kinda scenario, when the post get created what happens? If the post is not created what happens?

You can do this by having an If Else statement like below: 

```
if @post.save
do something
else
do something else
end
```

Reply:
***Hello Reviewer!***
I have implemented the changes required previously and hopefully all of them.